installers/install.sh

```diff
@@ -99,32 +107,39 @@ if [ -z "$TMPDIR" ]; then
   TMPDIR="/tmp"
 fi
 
-# Determine the latest version to fetch.
-STATEURL="$BASE_INFO_URL?channel=$CHANNEL&source=install&platform=$OS"
-$FETCH $TMPDIR/info.json $STATEURL || exit 1
-
-# Parse info.
-VERSION=`cat $TMPDIR/info.json | sed -ne 's/.*"version":[ \t]*"\([^"]*\)".*/\1/p'`
 if [ -z "$VERSION" ]; then
-  error "Unable to retrieve the latest version number"
-  exit 1
+  # Determine the latest version to fetch.
+  STATEURL="$BASE_INFO_URL?channel=$CHANNEL&source=install&platform=$OS"
+  $FETCH $TMPDIR/info.json $STATEURL || exit 1
+
+  # Parse info.
+  VERSION=`cat $TMPDIR/info.json | sed -ne 's/.*"version":[ \t]*"\([^"]*\)".*/\1/p'`
+  if [ -z "$VERSION" ]; then
+    error "Unable to retrieve the latest version number"
+    exit 1
+  fi
+  SUM=`cat $TMPDIR/info.json | sed -ne 's/.*"sha256":[ \t]*"\([^"]*\)".*/\1/p'`
+  RELURL=`cat $TMPDIR/info.json | sed -ne 's/.*"path":[ \t]*"\([^"]*\)".*/\1/p'`
+  rm $TMPDIR/info.json
+
+else
+  RELURL="$CHANNEL/$VERSION/$OS-amd64/state-$OS-amd64-$VERSION$DOWNLOADEXT"
 fi
-SUM=`cat $TMPDIR/info.json | sed -ne 's/.*"sha256":[ \t]*"\([^"]*\)".*/\1/p'`
-RELURL=`cat $TMPDIR/info.json | sed -ne 's/.*"path":[ \t]*"\([^"]*\)".*/\1/p'`
-rm $TMPDIR/info.json
 
-# Fetch the latest version.
+# Fetch the requested or latest version.
 progress "Preparing Installer for State Tool Package Manager version $VERSION"
 STATEURL="$BASE_FILE_URL/$RELURL"
 ARCHIVE="$OS-amd64$DOWNLOADEXT"
-if ! $FETCH $TMPDIR/$ARCHIVE $STATEURL ; then
+$FETCH $TMPDIR/$ARCHIVE $STATEURL
+if [ $? -ne 0 -o \( "`echo $FETCH | grep -o 'curl'`" == "curl" -a -z "`file -b $TMPDIR/$ARCHIVE | grep -o 'data'`" \) ]; then
```

Comment:
I'm not sure I follow what is happening here. Could you add a comment explaining?

Reply:
Thanks for the reminder. I totally meant to do that.